dash/include/dash/util/IndexSequence.h

```diff
@@ -28,6 +29,20 @@ struct make_index_sequence<0, Is...>
   : index_sequence<Is...>
 { };
 
+template <
+  std::size_t    N,
+  std::size_t... Is >
+struct rev_index_sequence
+  : rev_index_sequence<N-1, Is..., N-1>
```

Comment:
Shouldn't that be `rev_index_sequence<N-1, Is..., N>`? (So that `rev_index_sequence<3, 2, 1>` boils down to `index_sequence<1, 2, 3>`?) But I'm just guessing here from the documentation, though...

Reply:
No. That is why I call it rev_index_sequence (reverse index sequence). ;)

Reply:
No. That is why I call it rev_index_sequence (reverse index sequence). ;)

Reply:
Ahh, I didn't fully grasp the meaning of `index_sequence` at first (I'll blame my lack of sleep). Some documentation might be useful though ;)

Also: `dash::detail::reverse` should be named `dash::detail::reverse_impl` for consistency. For the same reason, you might want to rename `rev_index_sequence` to `make_rev_index_sequence` but I have no strong feelings about that.

Reply:
Good point. will do that. :)